From f3c3e310a60f77486da0519b37489007d67f301a Mon Sep 17 00:00:00 2001
From: Yang Shi <yang.shi@windriver.com>
Date: Thu, 24 May 2012 16:06:48 -0700
Subject: [PATCH 239/248] lttng: Replace seqlock API to seqcount API for vsyscall_gtod_data

Due to commit 2ab516575f2f273b19d95140d02c54612201e80a
[x86: vdso: Use seqcount instead of seqlock],
adopt seqcount API to manipulate it.

Signed-off-by: Yang Shi <yang.shi@windriver.com>
---
 arch/x86/kernel/vsyscall_64.c  |    6 ++++--
 arch/x86/vdso/vclock_gettime.c |    4 ++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/arch/x86/kernel/vsyscall_64.c b/arch/x86/kernel/vsyscall_64.c
index f239a2e..4d00f2a 100644
--- a/arch/x86/kernel/vsyscall_64.c
+++ b/arch/x86/kernel/vsyscall_64.c
@@ -89,9 +89,11 @@ void update_trace_clock_is_sync_vdso(void)
 {
 	unsigned long flags;
 
-	write_seqlock_irqsave(&vsyscall_gtod_data.lock, flags);
+	local_irq_save(flags);
+	write_seqcount_begin(&vsyscall_gtod_data.seq);
 	vsyscall_gtod_data.trace_clock_is_sync = _trace_clock_is_sync;
-	write_sequnlock_irqrestore(&vsyscall_gtod_data.lock, flags);
+	write_seqcount_end(&vsyscall_gtod_data.seq);
+	local_irq_restore(flags);
 }
 EXPORT_SYMBOL_GPL(update_trace_clock_is_sync_vdso);
 
diff --git a/arch/x86/vdso/vclock_gettime.c b/arch/x86/vdso/vclock_gettime.c
index f127b0a..9cd4748 100644
--- a/arch/x86/vdso/vclock_gettime.c
+++ b/arch/x86/vdso/vclock_gettime.c
@@ -164,7 +164,7 @@ notrace static noinline int do_trace_clock(struct timespec *ts)
 	union lttng_timespec *lts = (union lttng_timespec *) ts;
 
 	do {
-		seq = read_seqbegin(&gtod->lock);
+		read_seqcount_begin(&gtod->seq);
 		if (unlikely(!gtod->trace_clock_is_sync))
 			return vdso_fallback_gettime(CLOCK_TRACE, ts);
 		/*
@@ -179,7 +179,7 @@ notrace static noinline int do_trace_clock(struct timespec *ts)
 		 * match the TSC read by get_cycles() at the kernel level.
 		 */
 		lts->lttng_ts = vget_cycles();
-	} while (unlikely(read_seqretry(&gtod->lock, seq)));
+	} while (unlikely(read_seqcount_retry(&gtod->seq, seq)));
 
 	return 0;
 }
-- 
1.7.0.4

